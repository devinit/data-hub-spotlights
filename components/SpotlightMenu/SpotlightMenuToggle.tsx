import classNames from 'classnames';
import React, { FunctionComponent } from 'react';

interface ComponentProps {
  show?: boolean;
  onClick?: () => void;
  caption: string;
}

const SpotlightMenuToggle: FunctionComponent<ComponentProps> = ({ show, caption, onClick }) => {
  return (
    <nav className={classNames('countries-menu-list js-countries-menu-trigger', { inactive: !show })}>
      <a className="countries-menu-list__item countries-menu-list__parent" onClick={onClick}>
        <span>{caption}</span>
        <style jsx>{`
          cursor: pointer;
          text-transform: capitalize;
        `}</style>
      </a>
    </nav>
  );
};

SpotlightMenuToggle.defaultProps = { show: false };

export { SpotlightMenuToggle };
