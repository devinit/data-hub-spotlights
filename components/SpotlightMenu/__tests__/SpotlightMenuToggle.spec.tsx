/**
 * @jest-environment jsdom
 */
import '@testing-library/jest-dom/extend-expect';
import React from 'react';
import * as TestRenderer from 'react-test-renderer';
import { SpotlightMenuToggle } from '../SpotlightMenuToggle';

describe('SpotlightMenuToggle', () => {
  test('renders correctly', () => {
    const renderer = TestRenderer.create(<SpotlightMenuToggle caption="Uganda" />).toJSON();

    expect(renderer).toMatchSnapshot();
  });

  test('renders children correctly', () => {
    const renderer = TestRenderer.create(
      <SpotlightMenuToggle caption="Uganda">
        <div>Testing</div>
      </SpotlightMenuToggle>
    ).toJSON();

    expect(renderer).toMatchSnapshot();
  });

  test('It adds inactive class when show prop is false', () => {
    const renderer = TestRenderer.create(<SpotlightMenuToggle caption="Uganda" show={false} />).toJSON();

    expect(renderer).toMatchSnapshot();
  });
});
