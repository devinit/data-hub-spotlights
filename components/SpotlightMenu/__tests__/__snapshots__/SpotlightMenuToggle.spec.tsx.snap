// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SpotlightMenuToggle It adds inactive class when show prop is false 1`] = `
<nav
  className="countries-menu-list js-countries-menu-trigger inactive"
>
  <a
    className="countries-menu-list__item countries-menu-list__parent"
  >
    <span>
      Uganda
    </span>
    <style
      jsx={true}
    >
      
          cursor: pointer;
        
    </style>
  </a>
</nav>
`;

exports[`SpotlightMenuToggle It adds inactive class when show prop is true 1`] = `
<nav
  className="countries-menu-list js-countries-menu-trigger"
>
  <a
    className="countries-menu-list__item countries-menu-list__parent"
  >
    <span>
      Uganda
    </span>
    <style
      jsx={true}
    >
      
          cursor: pointer;
        
    </style>
  </a>
</nav>
`;

exports[`SpotlightMenuToggle renders children correctly 1`] = `
<nav
  className="countries-menu-list js-countries-menu-trigger inactive"
>
  <a
    className="countries-menu-list__item countries-menu-list__parent"
  >
    <span>
      Uganda
    </span>
    <style
      jsx={true}
    >
      
          cursor: pointer;
        
    </style>
  </a>
</nav>
`;

exports[`SpotlightMenuToggle renders correctly 1`] = `
<nav
  className="countries-menu-list js-countries-menu-trigger inactive"
>
  <a
    className="countries-menu-list__item countries-menu-list__parent"
  >
    <span>
      Uganda
    </span>
    <style
      jsx={true}
    >
      
          cursor: pointer;
        
    </style>
  </a>
</nav>
`;
